### PR TITLE
[IMP] delivery_driver_stock_picking_batch: Show only name in drivers

### PR DESCRIPTION
**Pull request by EmilioPascual**
- Assign the driver's user in batch if batch isn't wave.
- Show only name in drivers

PR Series: https://github.com/OCA/delivery-carrier/pull/774 (must be merged before this one)

MT-5232 @moduon

@shide @rafaelbn please review

### CODE REVIEW

File: delivery_driver_stock_picking_batch/models/stock_picking_batch.py [hunk 1]
```diff
@@ -15,8 +15,19 @@ class StockPickingBatch(models.Model):
...
+        store=True,
+    )
 
     @api.depends("picking_ids.driver_id")
     def _compute_driver_ids(self):
         for batch in self:
             batch.driver_ids = batch.picking_ids.mapped("driver_id")
+
+    @api.depends("driver_ids")
+    def _compute_user_id(self):
```
Comment (rousseldenis) [inline delivery_driver_stock_picking_batch/models/stock_picking_batch.py]: I'm not sure at all this is correct as this modifies the user_id values for all flows even without drivers.

IMHO, this should be readonly=False too.

File: delivery_driver_stock_picking_batch/models/stock_picking_batch.py [hunk 2]
```diff
@@ -15,8 +15,19 @@ class StockPickingBatch(models.Model):
...
+        for batch in self.filtered(lambda b: not b.is_wave):
+            users = batch.mapped("driver_ids.user_ids")
+            if users:
+                batch.user_id = users[0]
```
Comment (rousseldenis) [inline delivery_driver_stock_picking_batch/models/stock_picking_batch.py]: So, the user is the first driver if several mentionned on all pickings of the batch ??

**Comment by rafaelbn** [comment]
Example:

- We have our own 16 trucks for delivery
- We have our clients that sometime wants more and sometimes less, so every client has its delivery method.
- But if a Delivery method has few kilos, the we join in the same batch different delivery method.
- One (1) single drivers make that in 1 truck. 
- Automatically assigned is the first but this could be changed later.

😄

File: delivery_driver_stock_picking_batch/models/stock_picking_batch.py (hunk 1 above)
Comment (rafaelbn) [reply delivery_driver_stock_picking_batch/models/stock_picking_batch.py]: @rousseldenis do you know other module that computes this? If driver is not set or this module is not installed this don't impact

File: delivery_driver_stock_picking_batch/models/stock_picking_batch.py (hunk 2 above)
Comment (rafaelbn) [reply delivery_driver_stock_picking_batch/models/stock_picking_batch.py]: Only one driver can drive a batch but as commented here: https://github.com/OCA/delivery-carrier/pull/775#issuecomment-1992038756 some times you want to join for get the maximun weight per truck so even initially multi-drivers should work, in this cases only one of them will make the delivery

**Comment by EmilioPascual** [comment]
@rafaelbn although you are right, I think the functionality may be controversial and not suitable for all workflows. I have decided to move the functionality to our repository.

**Comment by EmilioPascual** [comment]
Failed tests are for another module: https://github.com/OCA/delivery-carrier/actions/runs/8263936917/job/22606437935?pr=775#step:8:335

**Review by rafaelbn — APPROVED** [review]
Ready to merge in my side 👍🏼 

> @rafaelbn although you are right, I think the functionality may be controversial and not suitable for all workflows. I have decided to move the functionality to our repository.

Perfect.

**Review by Shide — APPROVED** [review]

**Pull request merged**